src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int INITIAL_VALUE = 0;
+    private static final int OPERATION_PART = 0;
+    private static final int TRANSACTION_AMOUNT_PART = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supply = INITIAL_VALUE;
+        int buy = INITIAL_VALUE;
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String csvData;
+
+            while ((csvData = bufferedReader.readLine()) != null) {
+                String[] dataParts = csvData.split(",");
+                String operation = dataParts[OPERATION_PART];
+                int transactionAmount = Integer.parseInt(dataParts[TRANSACTION_AMOUNT_PART]);
+
+                if (operation.equals("supply")) {
+                    supply += transactionAmount;
+                } else if (operation.equals("buy")) {
+                    buy += transactionAmount;
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found" + fromFileName);
+        } catch (IOException e) {
+            System.out.println("Can`t read file" + fromFileName);
+        }
+
+        int result = supply - buy;
+        StringBuilder reportBuilder = new StringBuilder();
+        reportBuilder.append("supply,").append(supply).append(System.lineSeparator())
+                .append("buy,").append(buy).append(System.lineSeparator())
+                .append("result,").append(result);
```

This catch block prints the write error instead of rethrowing it. The checklist says: "Don't ignore exceptions". Replace `System.out.println("Can`t write to file" + toFileName);` with `throw new RuntimeException("Can't write data to file " + toFileName, e);` so callers/tests receive the error and can fail loudly.